src/seedu/addressbook/parser/Parser.java

```diff
@@ -301,7 +343,56 @@ private Command prepareEmployeeDelete(String args) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EmployeeDeleteCommand.MESSAGE_USAGE));
         }
     }
-    
+
+    /**
+     * Parses arguments in the context of the edit employee command.
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareEmployeeEdit(String args) {
+        final Matcher matcher = EMPLOYEE_EDIT_DATA_ARGS_FORMAT.matcher(args.trim());
+        if (!matcher.matches()) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EmployeeEditCommand.MESSAGE_USAGE));
+        }
+        try{
+            final int targetIndex = parseArgsAsDisplayedIndex(matcher.group("targetIndex"));
+            return new EmployeeEditCommand(
+                    targetIndex,
+                    checkEmpty(matcher.group("phone"), "phone"),
+                    checkEmpty(matcher.group("email"), "email"),
+                    checkEmpty(matcher.group("address"), "address"),
+                    checkEmpty(matcher.group("position"), "position")
+            );
+        } catch (ParseException | NumberFormatException e) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EmployeeEditCommand.MESSAGE_USAGE));
+        } catch (IllegalValueException ive) {
+        return new IncorrectCommand(ive.getMessage());
+        }
+    }
+
+    /**
+     * Returns new information to be edited if it is not empty
+     */
+    private static String checkEmpty(String toCheck, String argument){
```

Comment:
Method name suggests that the method is a boolean method, so it should be changed

Reply:
Edited.